### PR TITLE
Update Readme to document hub and managed cluster setup

### PR DESCRIPTION
**Pull request by sacharya**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,36 @@
 
 This operator provides a CRD that is used to hold inventory records in the hub cluster, and a controller that reconciles inventory with resources in the managed cluster. The actual placement of resources from hub to managed cluster is done using either hive's SyncSet or MCM.
```
Comment (pawanpinjarkar) [inline README.md]: ## Pre-requisites

- OC CLI - https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/latest-4.3/
- mockgen - GO111MODULE=on go get github.com/golang/mock/mockgen@latest
- kustomize - GO111MODULE=on go get sigs.k8s.io/kustomize/kustomize/v3@latest
- cfssl and cfssljson 
```
$ sudo curl https://pkg.cfssl.org/R1.2/cfssl_linux-amd64 -o /usr/local/bin/cfssl
$ sudo curl https://pkg.cfssl.org/R1.2/cfssljson_linux-amd64 -o /usr/local/bin/cfssljson
$ sudo chmod +x /usr/local/bin/cfssl /usr/local/bin/cfssljson
```
- Golint - GO111MODULE=on go get -u golang.org/x/lint/golint
```
# binary will be $(go env GOPATH)/bin/golangci-lint
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.23.1
# or install it into ./bin/
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.23.1
# In alpine linux (as it does not come with curl by default)
wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.23.1
golangci-lint --version
```

File: README.md [hunk 2]
```diff
@@ -2,6 +2,36 @@
...
+cd hive
+make deploy
+
+rm -rf hiveadmission-certs
+./hack/hiveadmission-dev-cert.sh
+```
+
+Adopt cluster0 as a managed cluster
+```
+bin/hiveutil create-cluster --adopt --adopt-admin-kubeconfig=/tmp/cluster0.config --adopt-cluster-id=cluster0 --adopt-infra-id=cluster0 cluster0
```
Comment (pawanpinjarkar) [inline README.md]: Also, add ```--include-secrets=false ``` to skip aws credentials requirement

File: README.md [hunk 3]
```diff
@@ -2,6 +2,36 @@
...
+Note: If the kubeconfig file includes keys, like certificate-authority, client-certificate and client-key, with reference to local files, replace them with certificate-authority-data, client-certificate-data and client-key-data.
+
+## Start a hub cluster named hive
+```
+minikube start -p hive
+```
+
+Deploy hive to the hub cluster.
+```
+cd $GOPATH/src/github.com/openshift
```
[the remark this answers is not in the document]
Comment (sacharya) [reply README.md]: Removed the detail altogether and just linked to hive docs, which is what we should be following.

File: README.md [hunk 4]
```diff
@@ -2,6 +2,36 @@
...
+git clone https://github.com/openshift/hive
+cd hive
+make deploy
+
+rm -rf hiveadmission-certs
+./hack/hiveadmission-dev-cert.sh
+```
+
+Adopt cluster0 as a managed cluster
+```
```
[the remark this answers is not in the document]
Comment (sacharya) [reply README.md]: Follow hive docs.

File: README.md (hunk 1 above)
Comment (sacharya) [reply README.md]: Hive docs should cover that.

File: README.md (hunk 2 above)
Comment (sacharya) [reply README.md]: Added a note.

File: README.md [hunk 5]
```diff
@@ -2,6 +2,36 @@
 
 This operator provides a CRD that is used to hold inventory records in the hub cluster, and a controller that reconciles inventory with resources in the managed cluster. The actual placement of resources from hub to managed cluster is done using either hive's SyncSet or MCM.
 
+## Start a managed cluster named cluster0
+```
+minikube start -p cluster0
+kubectl apply -f https://raw.githubusercontent.com/metal3-io/baremetal-operator/master/deploy/crds/metal3.io_baremetalhosts_crd.yaml
+```
+
+Save cluster0's kubeconfig to /tmp/cluster0.config.
```
[the remark this answers is not in the document]
Comment (sacharya) [reply README.md]: Removed it.  --adopt-admin-kubeconfig=/tmp/cluster0.config

**Pull request merged**